crates/edr_provider/tests/issue_384.rs

```diff
@@ -0,0 +1,27 @@
+#![cfg(feature = "test-utils")]
+
+use edr_provider::{
+    hardhat_rpc_types::ForkConfig, test_utils::create_test_config_with_fork, MethodInvocation,
+    NoopLogger, Provider, ProviderRequest,
+};
+use tokio::runtime;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn flare_network_mine_local_block() -> anyhow::Result<()> {
+    const BLOCK_NUMBER: u64 = 22_587_773;
+
+    let logger = Box::new(NoopLogger);
+    let subscriber = Box::new(|_event| {});
+
+    let config = create_test_config_with_fork(Some(ForkConfig {
+        json_rpc_url: "https://flare-api.flare.network/ext/bc/C/rpc".to_string(),
```

Comment:
I think it would be better to do this, assuming our Infura endpoint support Avalanche:

```
        json_rpc_url: get_infura_url().replace("mainnet", "avalanche-mainnet"),
```

Public endpoints are more likely to fail, and that would generate false negatives for us.

Reply:
I don't have access to the Infura token used by the CI, so the simplest thing here is to just try it out and see if the CI passes :man_shrugging:

Reply:
Thanks! Addressed by https://github.com/NomicFoundation/hardhat/pull/5127/commits/3f2390478c1004095673e9e2e658cee486f0e64b

TIL that Flare and Avalanche have common properties 😁